io.github.Dretch.MonomerFlatpakExample.json

```diff
@@ -0,0 +1,3423 @@
+        "runhaskell Setup build",
+        "runhaskell Setup install"
+      ],
+      "builddir": false,
+      "buildsystem": "simple",
+      "name": "monomer",
+      "sources": [
+        {
+          "sha256": "1169c03dfb65fa1347c2f417b341249a65e07a310dd1ce1b4a9934e485a6e68f",
+          "type": "archive",
+          "url": "https://hackage.haskell.org/package/monomer-1.5.1.0/monomer-1.5.1.0.tar.gz"
+        },
+        {
+          "dest-filename": "monomer.cabal",
+          "sha256": "18e3ce60c7f5d3fae74bf5ace7515f0e252b12ccc4b71030f948e6fdc1f49a99",
+          "type": "file",
+          "url": "https://hackage.haskell.org/package/monomer-1.5.1.0/revision/0.cabal"
+        }
+      ]
+    },
+    {
+      "build-commands": [
+        "echo '#! /usr/bin/env runhaskell' >Setup.txt",
+        "echo '> import Distribution.Simple' >>Setup.txt",
+        "echo '> main = defaultMain' >>Setup.txt",
+        "ln -s Setup.txt Setup.lhs || true",
+        "runhaskell Setup configure --prefix=/app",
+        "runhaskell Setup build",
+        "runhaskell Setup install"
+      ],
+      "builddir": false,
+      "buildsystem": "simple",
+      "name": "monomer-hagrid",
+      "sources": [
+        {
+          "sha256": "82cede73306f9b035425c2748acf369a706a0afcaf502474a363dffbefc9bd2c",
+          "type": "archive",
+          "url": "https://hackage.haskell.org/package/monomer-hagrid-0.2.1.1/monomer-hagrid-0.2.1.1.tar.gz"
+        },
+        {
+          "dest-filename": "monomer-hagrid.cabal",
+          "sha256": "29245a958533b86c5b87004fd767c72871795ce2fd598417c388a1402fb9f518",
+          "type": "file",
+          "url": "https://hackage.haskell.org/package/monomer-hagrid-0.2.1.1/revision/0.cabal"
+        }
+      ]
+    },
+    {
+      "build-commands": [
+        "echo '#! /usr/bin/env runhaskell' >Setup.txt",
+        "echo '> import Distribution.Simple' >>Setup.txt",
+        "echo '> main = defaultMain' >>Setup.txt",
+        "ln -s Setup.txt Setup.lhs || true",
+        "runhaskell Setup configure --prefix=/app",
+        "runhaskell Setup build",
+        "runhaskell Setup install"
+      ],
+      "builddir": false,
+      "buildsystem": "simple",
+      "name": "monomer-flatpak-example",
+      "sources": [
+        {
+          "sha256": "4f5ff54bc492d5be9459ed1542151e0d8db69a44f2daf4b8caacbd79098b1f11",
+          "type": "archive",
+          "url": "https://hackage.haskell.org/package/monomer-flatpak-example-0.0.2.6/monomer-flatpak-example-0.0.2.6.tar.gz"
+        },
+        {
+          "dest-filename": "monomer-flatpak-example.cabal",
+          "sha256": "608b184cca1d2367d38eacf85e8e951659efe8f92333d84c40976a600f994441",
+          "type": "file",
+          "url": "https://hackage.haskell.org/package/monomer-flatpak-example-0.0.2.6/revision/0.cabal"
+        }
+      ]
+    },
+    "install-desktop-files.json"
+  ],
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
+  "sdk": "org.freedesktop.Sdk"
```

Comment:
these are usually at the top of the manifest. Logical order. Otherwise it's nearly impossible to read.

Reply:
The dependencies that are auto-generated should be in a separate manifest so it can be re-generated without changing the main manifest. This is what is done when using the flatpak-builder-tools (python, cargo, node, etc)

As for the tool, well I wrote my own flaptak manifest generator to bootstrap packages and I made sure it kept the keys in order of insertiion.